### PR TITLE
up to vlm part

### PR DESCRIPTION
**Pull request by deep-diver**
this program assumes that you are authorized. you can test out after 
```shell
$ gcloud auth application-default login
```

will update the README.md about `gcloud` and stuff about `gcp` later. after running this program, got the following output as Python dictionary (parsing done).

```
{'characters': [{'name': 'Lily', 'physicalDescription': 'Lily is a young woman in her early twenties with long, wavy brown hair and hazel eyes. She is of average height and has a slim build. She is often seen wearing a red beanie and a yellow raincoat.', 'personalityTraits': ['carefree', 'spontaneous', '乐观', 'friendly', 'creative'], 'likes': ['taking pictures', 'exploring new places', 'spending time with friends', 'music', 'art'], 'dislikes': ['being confined', 'mundane routines', 'injustice', 'cruelty', 'disorganization'], 'background': 'Lily is a recent college graduate who is trying to find her place in the world. She is originally from a small town in the Midwest, but she moved to the city to pursue her dreams of becoming a photographer. She is currently working as a freelance photographer and is hoping to build a successful career in the field.', 'goals': ['to become a successful photographer', 'to travel the world', 'to make a difference in the world', 'to find love and happiness'], 'relationships': [{'name': 'Sam', 'relation': 'best friend', 'description': "Sam is Lily's best friend. They met in college and have been inseparable ever since. Sam is always there for Lily, no matter what. She is a great listener and always offers sound advice. Lily can always count on Sam to make her laugh and have a good time."}, {'name': 'Jack', 'relation': 'boyfriend', 'description': "Jack is Lily's boyfriend. They have been dating for about a year and are very happy together. Jack is kind, caring, and supportive. He loves Lily very much and is always there for her. Lily is very lucky to have Jack in her life."}]}]}
```

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+def workflow(args):
+    traits = vlm.get_traits(
+        args.gcp_project_id, 
+        args.gcp_project_location, 
+        args.target_movie_clip, 
+        args.prompt_tpl_path
+    )
+    print(traits)
+
+def sanity_check(args):
```
Comment (sayakpaul) [inline main.py]: `validate_inputs` might be a better name here. Also WDYT about doing this using a decorator, responsible for validating the necessary inputs? We could decorate the `workflow()` function then.

File: vid2persona/gen/gemini.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+def _default_gen_config():
+    return GenerationConfig(
+        max_output_tokens=2048,
+        temperature=0.4,
+        top_p=1,
+        top_k=32
+    )
+
+def _default_model_name():
+    return "gemini-1.0-pro-vision"
```
Comment (sayakpaul) [inline vid2persona/gen/gemini.py]: This could simply be set as a CONSTANT no?

File: vid2persona/gen/tgi_openllm.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
+import json
```
Comment (sayakpaul) [inline vid2persona/gen/tgi_openllm.py:1]: I don't know what this is doing.

**Review by sayakpaul — COMMENTED** [review]
Nice start. I have left a couple of comments. Major point is I don't understand what `tgi_openllm.py` is doing exactly. 

Additionally, we could maybe use [this](https://gist.github.com/sayakpaul/a48a0f63d0e346164a97c687f27c5b51) for formatting the project. But this is just an opinion. Doesn't mean we have to implement it.

File: main.py (hunk 1 above)
Comment (deep-diver) [reply main.py]: Never thought about decorator! Good suggestion!

File: vid2persona/gen/gemini.py (hunk 2 above)
Comment (deep-diver) [reply vid2persona/gen/gemini.py]: Yeah it is! Just wanted to make it function for future usage. But yeah I agree with you

File: vid2persona/gen/tgi_openllm.py (hunk 3 above)
Comment (deep-diver) [reply vid2persona/gen/tgi_openllm.py:1]: Just ignore it for now. I was about to test the pipeline with TGI, but didn't have enough time

**Comment by deep-diver** [comment]
@sayakpaul 

- `tgi_openllm.py` is my approach to implement TGI client for HF PRO users. I just placed this since this code snippet is what I feel familiar with (so just fast prototyping). But we can replace this with OpenAI SDK or TGI client library. Please just think it as a placeholder for now!

- thanks for suggesting to use `ruff`! I am not super familiar with `ruff` yet, so probably I will be sticking with the current structure. Let me play with it for a couple of days, then if I feel comfortable, let me open another PR to replace the code base structure.

What do you think?

**Comment by sayakpaul** [comment]
All good!

**Pull request merged**